drivers/ethernet/eth_sam_gmac.c

```diff
@@ -526,16 +541,27 @@ static void tx_completed(Gmac *gmac, struct gmac_queue *queue)
 static void tx_error_handler(Gmac *gmac, struct gmac_queue *queue)
 {
 	struct net_pkt *pkt;
+	struct net_buf *frag;
+	struct ring_buf *tx_frag_list = &queue->tx_frag_list;
 	struct ring_buf *tx_frames = &queue->tx_frames;
 
 	queue->err_tx_flushed_count++;
 
 	/* Stop transmission, clean transmit pipeline and control registers */
 	gmac->GMAC_NCR &= ~GMAC_NCR_TXEN;
 
+	/* Free all frag resources in the TX path */
+	while (tx_frag_list->tail != tx_frag_list->head) {
+		/* Release net buffer to the buffer pool */
+		frag = UINT_TO_POINTER(tx_frag_list->buf[tx_frag_list->tail]);
+		net_buf_unref(frag);
```

Comment:
net_pkt_frag_unref() (see my comment about net_pkt_frag_ref )

Reply:
Ok, I'll fix that

Reply:
I have just pushed a version fixing that.